src/main/java/com/aws/greengrass/deployment/DeploymentDocumentDownloader.java

```diff
@@ -202,12 +199,14 @@ private void validateHttpExecuteResponse(HttpExecuteResponse executeResponse)
         }
     }
 
-    private Configuration deserializeDeploymentDoc(String configurationInString)
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private DeploymentDocument deserializeDeploymentDoc(String configurationInString)
             throws DeploymentTaskFailureException {
         try {
-            return SerializerFactory.getFailSafeJsonObjectMapper()
+            Configuration configuration =  SerializerFactory.getFailSafeJsonObjectMapper()
                     .readValue(configurationInString, Configuration.class);
-        } catch (JsonProcessingException e) {
+            return DeploymentDocumentConverter.convertFromDeploymentConfiguration(configuration);
+        } catch (Exception e) {
```

Comment:
I don't like using generic exception. Would prefer JsonProcessingException separate and InvalidRequestException separate.

Reply:
Discussed with Amit offline, This adds more context to the error message. Other wise the unhandled errors  gets propagated to Deployment service and would be surfaced to the customers (Like NPE). Catching it here and adding more context will make it more readable.